Loop/View Controllers/CarbEntryViewController.swift

```diff
@@ -216,23 +216,25 @@ final class CarbEntryViewController: LoopChartsTableViewController, Identifiable
     private var foodKeyboard: EmojiInputController!
     
     private func updateDisplayAccurateCarbEntryWarning() {
-        deviceManager.loopManager.getLoopState { [weak self] (_, state) in
-            let endDate = Date()
-            let startDate = endDate.addingTimeInterval(.minutes(-20))
-            let threshold = HKQuantity(unit: GlucoseEffectVelocity.unit, doubleValue: 3)
-
-            let filteredInsulinCounteractionEffects = state.insulinCounteractionEffects.filterDateRange(startDate, endDate)
+        let now = Date()
```

Comment:
I'd really love to see such policy code be put into a utility (or somewhere equivalent) and have unit tests written.

Reply:
Yeah, I would like to refactor this to make it more testable; i.e. make some simpler protocols for the device manager dependency and make the meal detection a separate model, but I figured for this timeline, I'd keep the existing pattern and we can look at a refactor later.